go/v1/statement_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+ * Tests for in-toto attestation ResourceDescriptor protos.
+*/
+
+package v1
+
+import (
+	"testing"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/structpb"
+	"github.com/stretchr/testify/assert"
+)
+
+const wantSt = `{"_type":"https://in-toto.io/Statement/v1","subject":[{"name":"theSub","digest":{"alg1":"abc123"}}],"predicateType":"thePredicate","predicate":{"keyObj":{"subKey":"subVal"}}}`
+
+func createTestStatement() (*Statement, error) {
+	// Create a Statement
+	sub, err := NewResourceDescriptorPb("theSub", "",
+		map[string]string{"alg1": "abc123"}, nil, "", "", nil)
+	if err != nil {
+		return nil, err
+	}
+	
+	pred, err := structpb.NewStruct(map[string]interface{}{
+		"keyObj": map[string]interface{}{
+			"subKey": "subVal"}})
+	if err != nil {
+		return nil, err
+	}
+
+	return NewStatementPb([]*ResourceDescriptor{sub}, "thePredicate",
+		pred)
+}
+
+func TestJsonUnmarshalStatement(t *testing.T) {	
+	got := &Statement{}
+	err := protojson.Unmarshal([]byte(wantSt), got)
+
+	assert.Nil(t, err, "Error during JSON unmarshalling")
+
+	want, err := createTestStatement()
+
+	assert.Nil(t, err, "Error during test Statement creation")
```

Comment:
nitpick, you might be able to use

`assert.Nil` => `assert.NoError`  https://pkg.go.dev/github.com/stretchr/testify/assert#NoError

Reply:
Yep, thanks!